applications/FluidDynamicsApplication/custom_utilities/fluid_auxiliary_utilities.cpp

```diff
@@ -418,4 +420,73 @@ void FluidAuxiliaryUtilities::CalculateSplitConditionGeometryData<false>(
     rpModShapeFunc->ComputeNegativeExteriorFaceAreaNormals(rNormals, FaceId, GeometryData::IntegrationMethod::GI_GAUSS_2);
 }
 
+    void FluidAuxiliaryUtilities::EmbeddedSkinToVolumeMapping(
+        ModelPart& rVolumeModelPart,
+        ModelPart& rSkinModelPart,
+        const double SearchRadius
+        )
+    {
+        //put the nodes in the skin into a search database
+        typedef BinsDynamic<3, Node<3>, std::vector<Node<3>::Pointer>> DynamicBins;
+        DynamicBins locator(rSkinModelPart.Nodes().GetContainer().begin(), rSkinModelPart.Nodes().GetContainer().end());
+
+        //reset the EMBEDDED_VELOCITY
+        block_for_each(rVolumeModelPart.Nodes(), [](auto& rNode){
+            rNode.GetValue(EMBEDDED_VELOCITY).clear();
+        });
+
+        //detect all the nodes that belong to cut elements
+        ModelPart::NodesContainerType cut_elem_nodes;
+        for(const auto& r_elem : rVolumeModelPart.Elements()){
+            const auto& r_geom = r_elem.GetGeometry();
+            
+            unsigned int npos=0, nneg=0;
+            for (const auto& r_node : r_geom) {
+                r_node.FastGetSolutionStepValue(DISTANCE) > 0 ? ++npos : ++nneg;
+            }
+
+            if(npos>0 && nneg>0){ //element is split
+                for(unsigned int i=0; i<r_geom.size(); ++i) {
+                    cut_elem_nodes.push_back(r_geom(i));
+                }
+            }
+        }
+        cut_elem_nodes.Unique();
+
+        unsigned int max_results = 100; //deliberately low, we do not want to interpolate from more than this nodes
+        auto TLS = std::make_pair(std::vector<Node<3>::Pointer>(max_results), std::vector<double>(max_results));
+        
+        //for every interface node (nodes in cut elements)
+        block_for_each(cut_elem_nodes, TLS, [&locator, SearchRadius](auto& rNode, auto& rTLS){
+                //find all neighbours within a radius
+                auto& r_neighbours = rTLS.first;
+                auto& r_distances  = rTLS.second;
+                const unsigned int nfound = locator.SearchInRadius(
+                    rNode, 
+                    SearchRadius, 
+                    r_neighbours.begin(), 
+                    r_distances.begin(), 
+                    r_neighbours.size());
+                KRATOS_ERROR_IF(nfound == 0) << "Not found any neighbor for node " << rNode.Id() << " in skin model part." << std::endl;
+
+                //compute RBF basis
+                Matrix cloud_coords(nfound,3);
+                Vector shape_functions(nfound);
+                for(unsigned int i=0; i<nfound; ++i){
+                    cloud_coords(i,0)=r_neighbours[i]->X(); 
+                    cloud_coords(i,1)=r_neighbours[i]->Y(); 
+                    cloud_coords(i,2)=r_neighbours[i]->Z();
+                }
+                RBFShapeFunctionsUtility::CalculateShapeFunctions(cloud_coords,rNode.Coordinates(), shape_functions);
```

Comment:
hm this is like half of an RBF mapper, maybe I should use this to implement it ...

Reply:
well, more than half...it IS a RBF mapping...except that we do not return the interpolation matrix but just the application of it to some data. (with the new matrix returning T would be quite easy)

Reply:
I think there is a second step required to compute the coefficients for the T matrix

Reply:
the coerficients of T are the shape functions...

Reply:
Hmm I have to check

Will get back to you :D

Reply:
maybe we should have a short conversation about this. send me a telegram when you have some time